test/rules/promise-async/test.ts.lint

```diff
@@ -0,0 +1,38 @@
+const nonAsyncPromiseFunction = function(p: Promise<void>) { return p; };
```

Comment:
Got it, thanks. For this PR, mind if I just put `declare interface Promise<T>;` at the top of the test file? That at least seems to resolve the immediate issue.